Tools/VacuumTool/VacuumTool.cs

```diff
@@ -23,38 +28,77 @@ sealed class VacuumTool : MonoBehaviour, ITool, ICustomActionMap, IUsesRayOrigin
 
 		public Vector3 defaultOffset { private get; set; }
 		public Quaternion defaultTilt { private get; set; }
+		public Node? node { private get; set; }
+
+		void Start()
+		{
+			InputUtils.GetBindingDictionaryFromActionMap(m_ActionMap, m_Controls);
+		}
+
+		void OnDestroy()
+		{
+			this.ClearFeedbackRequests();
+		}
 
 		public void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeControl)
 		{
-			var vacuumInput = (VacuumInput)input;
-			if (vacuumInput.vacuum.wasJustPressed)
+			var hit = false;
+			foreach (var vacuumable in vacuumables)
 			{
-				var realTime = Time.realtimeSinceStartup;
-				if (UIUtils.IsDoubleClick(realTime - m_LastClickTime))
+				var vacuumableTransform = vacuumable.transform;
+				var ray = new Ray(rayOrigin.position, rayOrigin.forward);
+				ray.origin = vacuumableTransform.InverseTransformPoint(ray.origin);
+				ray.direction = vacuumableTransform.InverseTransformDirection(ray.direction);
+				if (vacuumable.vacuumBounds.IntersectRay(ray))
 				{
-					var hit = false;
-					foreach (var vacuumable in vacuumables)
+					hit = true;
+					var vacuumInput = (VacuumInput)input;
+					if (vacuumInput.vacuum.wasJustPressed)
 					{
-						var vacuumableTransform = vacuumable.transform;
-						var ray = new Ray(rayOrigin.position, rayOrigin.forward);
-						ray.origin = vacuumableTransform.InverseTransformPoint(ray.origin);
-						ray.direction = vacuumableTransform.InverseTransformDirection(ray.direction);
-						if (vacuumable.vacuumBounds.IntersectRay(ray))
+						var realTime = Time.realtimeSinceStartup;
+						if (UIUtils.IsDoubleClick(realTime - m_LastClickTime))
 						{
-							hit = true;
 							Coroutine coroutine;
 							if (m_VacuumingCoroutines.TryGetValue(vacuumableTransform, out coroutine))
 								StopCoroutine(coroutine);
 
 							m_VacuumingCoroutines[vacuumableTransform] = StartCoroutine(VacuumToViewer(vacuumable));
+							consumeControl(vacuumInput.vacuum);
+						}
+
+						m_LastClickTime = realTime;
+					}
+
+					if (m_Feedback.Count == 0)
+					{
+						foreach (var kvp in m_Controls)
+						{
+							foreach (var id in kvp.Value)
+							{
+								var request = new ProxyFeedbackRequest
+								{
+									control = id,
+									node = node.Value,
```

Comment:
Null check node before retrieving/assigning node.Value.  Looks like nulls can't occur in the current arch, but wanted to mention it.

Reply:
Yeah. My other thought was just to make the field a Node? so it can just be passed directly through. Otherwise that's a lot of null checks (here and in all the ShowXRequest methods in LocomotionTool).

@AndrewTHEManeri thoughts?

Reply:
Is there expected behavior for objects if they encounter a null node?  And if a 'null' value is expected, should the node enum itself just get an extra value?

Reply:
Yeah, a null node is meant to indicate that there is no hand/node involved. I'm also in favor of a "None" value for Node, and getting rid of all the Node?s

Reply:
I'm for "None", in order to lose the Null.